webrender/src/resource_cache.rs

```diff
@@ -483,25 +498,46 @@ impl ResourceCache {
                     return;
                 }
 
+                if let Some(tile_size) = template.tiling {
+                    // If the tiling size is too big for hardware texture size then
+                    // just early out and drop the image.
+                    if tile_size as u32 > self.texture_cache.max_texture_size() {
+                        warn!("Dropping image, tile size:{} is too big for hardware!", tile_size);
+                        self.cached_images.insert(request, Err(ResourceClassCacheError::OverLimitSize));
+                        return;
+                    }
+                } else {
+                    // The image is too big for hardware texture size.
+                    if template.descriptor.width > self.texture_cache.max_texture_size() ||
```

Comment:
should probably merge this block with the one above, using `let side_size = image.tiling.unwrap_or(cmp::max(template.descriptor.width, template.descriptor.height));`